subsurface-helper.cpp

```diff
@@ -147,55 +147,32 @@ static void register_meta_types()
 }
 #endif // not SUBSURFACE_TEST_DATA
 
+#define REGISTER_TYPE(useClass, useQML) \
+	rc = qmlRegisterType<useClass>("org.subsurfacedivelog.mobile", 1, 0, useQML); \
+	if (rc < 0) \
+		qDebug() << "ERROR: Cannot register useQML (class useClass), QML will not work!!";
+
```

Comment:
that's a good idea. Makes the code much easier to read.

Reply:
I don't think that works as intended (that's not how the parser / preprocessor tokenizes):

```
#include <cstdio>
#include <typeinfo>

#define TEST1 xyz
#define TEST2 "abc"

struct test
{
        int a;
};

int main()
{       
        test x, *y;
        printf("XYZ TEST1 XYZ\n");
        printf("ABC " TEST2 " ABC\n");
        printf("type-name of int: %s\n", typeid(int).name());
        printf("type-name of x: %s\n", typeid(x).name());
        printf("type-name of y: %s\n", typeid(y).name());
}
```
gives 
```
XYZ TEST1 XYZ
ABC abc ABC
type-name of int: i
type-name of x: 4test
type-name of y: P4test
```

Unfortunately, `typeid.name()` gives (on gcc) mangled names and "demangling" is not portable. On the other hand, for a debug message a mangled name might be enough.

So you could turn that into a template function (untested):
```
template <typename T>
void register(const char *useQML)
{
	if (qmlRegisterType<T>("org.subsurfacedivelog.mobile", 1, 0, useQML) < 0)
		qDebug() << QStringLiteral("ERROR: Cannot register %1 (class %2), QML will not work!!".arg(useQML, typeinfo(T).name();
}
```
And it might be a good idea to turn this from `qDebug` to `qWarning`.

Reply:
good catch, I forgot the parameter was inside “”. corrected